### ISSUE TITLE
PR #8997 breaks transitions for `Wall:Detailed`, etc for version < 9.6

### ISSUE THREAD

**Issue by hongyuanjia** [NREL/EnergyPlus#9172]
Issue overview
--------------
PR #8997 introduced some changes in the `VCompareGlobalRoutines.f90` for compatibility with the Space def introduced. However, those changes break transitions that involve detailed geometries, e.g. `Wall:Detailed`, `Floor:Detailed`, including transitions from v7.2 to v8.0, and v8.7 to v8.8.

I maintain an R package [eplusr](https://gihttps://github.com/hongyuanjia/eplusr) which implements pure R-based transition programs. Tests are run to make sure the results are the same between R and EnergyPlus Transition programs. The CI [gives me errors](https://github.com/hongyuanjia/eplusr/runs/4129690593?check_suite_focus=true) after I try to use EnergyPlus v9.6.

```
══ Failed tests ════════════════════════════════════════════════════════════════
── Failure (test-transition.R:199:5): Transition v7.2 --> v8.0 ─────────────────
idfVU$"Wall:Detailed"$Wall$value() not equal to idfTR$"Wall:Detailed"$Wall$value().
Length mismatch: comparison on first 39 components
── Failure (test-transition.R:204:5): Transition v7.2 --> v8.0 ─────────────────
idfVU$"RoofCeiling:Detailed"$Roof$value() not equal to idfTR$"RoofCeiling:Detailed"$Roof$value().
Length mismatch: comparison on first 39 components
── Failure (test-transition.R:209:5): Transition v7.2 --> v8.0 ─────────────────
idfVU$"Floor:Detailed"$Floor$value() not equal to idfTR$"Floor:Detailed"$Floor$value().
Length mismatch: comparison on first 39 components
── Failure (test-transition.R:1252:5): Transition v8.7 --> v8.8 ────────────────
idfVU$"Floor:Detailed"$Surf2$value() not equal to idfTR$"Floor:Detailed"$Surf2$value(1:21).
Length mismatch: comparison on first 21 components
Component "Vertex 1 Y-coordinate": Mean relative difference: 0.3777778
Component "Vertex 1 Z-coordinate": Mean relative difference: 1
Component "Vertex 2 X-coordinate": Mean absolute difference: 45
Component "Vertex 2 Y-coordinate": Mean relative difference: 0.9111111
Component "Vertex 2 Z-coordinate": Mean relative difference: 1
Component "Vertex 3 X-coordinate": Mean absolute difference: 4
Component "Vertex 3 Z-coordinate": Mean relative difference: 1
Component "Vertex 4 X-coordinate": Mean absolute difference: 4
...
```

Take the floor `Zn001:Flr001` as an example:
```
  Floor:Detailed,
    Zn001:Flr001,            !- Name
    FLOOR,                   !- Construction Name
    ZONE ONE,                !- Zone Name
    Adiabatic,               !- Outside Boundary Condition
    ,                        !- Outside Boundary Condition Object
    NoSun,                   !- Sun Exposure
    NoWind,                  !- Wind Exposure
    1.000000,                !- View Factor to Ground
    4,                       !- Number of Vertices
    15.24000,0.000000,0.0,  !- X,Y,Z ==> Vertex 1 {m}
    0.000000,0.000000,0.0,  !- X,Y,Z ==> Vertex 2 {m}
    0.000000,15.24000,0.0,  !- X,Y,Z ==> Vertex 3 {m}
    15.24000,15.24000,0.0;  !- X,Y,Z ==> Vertex 4 {m}
```
 After transitioning from v8.7 to v8.8 using VersionUpdater from EnergyPlus v9.6, the results are below. Note that the value of `Vertex 1 X-coordinate` has been **duplicated**.

```
  Floor:Detailed,
    Zn001:Flr001,            !- Name
    FLOOR,                   !- Construction Name
    ZONE ONE,                !- Zone Name
    Adiabatic,               !- Outside Boundary Condition
    ,                        !- Outside Boundary Condition Object
    NoSun,                   !- Sun Exposure
    NoWind,                  !- Wind Exposure
    1.000000,                !- View Factor to Ground
    4,                       !- Number of Vertices
    15.24000,                !- Vertex 1 X-coordinate {m}
    15.24000,0.000000,0.0,  !- X,Y,Z ==> Vertex 1 {m}
    0.000000,0.000000,0.0,  !- X,Y,Z ==> Vertex 2 {m}
    0.000000,15.24000,0.0,  !- X,Y,Z ==> Vertex 3 {m}
    15.24000,15.24000,0.0;  !- X,Y,Z ==> Vertex 4 {m}
```

The error comes from the wrong number of fields given in `WriteOutPartialIDFLines` after parsing the field `Number of Vertices`.

https://github.com/NREL/EnergyPlus/blob/00d968c57b48ff354fdc123d00e782e18842f1ba/src/Transition/VCompareGlobalRoutines.f90#L947-L972

Instead of giving a fixed **`10`**, the number of fields to be written should be based on `NvertFiledNum`. Just as what has been done to the case for `BuildingSurface:Detailed`. So L971 should be changed to below:

```
CALL WriteOutPartialIDFLines(DifUnit,ObjectName, **NVertFieldNum**,OutArgs,FieldNames,FieldUnits)
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus: Offical v9.6
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: [1ZoneUncontrolled_FoorDetailed.idf.txt](https://github.com/NREL/EnergyPlus/files/7491936/1ZoneUncontrolled_FoorDetailed.idf.txt)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
@jcyuan2020 Please add this fix to #9139.

**Comment by jcyuan2020** [comment]
This one is added too. Thanks @hongyuanjia @mjwitte

**Comment by mjwitte** [comment]
Addressed in #9139
